### ISSUE TITLE
[ci] CorruptedFileIT.testReplicaCorruption

### ISSUE THREAD

**Issue by andyb-elastic** [elastic/elasticsearch#32304]
Doesn't reproduce locally

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/2385/console

[build-2385.txt](https://github.com/elastic/elasticsearch/files/2221380/build-2385.txt)

```
REPRODUCE WITH: ./gradlew :server:integTest \
  -Dtests.seed=172540A0C77C8F60 \
  -Dtests.class=org.elasticsearch.index.store.CorruptedFileIT \
  -Dtests.method="testReplicaCorruption" \
  -Dtests.security.manager=true \
  -Dtests.locale=en-MT \
  -Dtests.timezone=Chile/EasterIsland
```

```
ERROR   5.17s J2 | CorruptedFileIT.testReplicaCorruption <<< FAILURES!
   > Throwable #1: com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=9856, name=elasticsearch[node_sd3][write][T#2], state=RUNNABLE, group=TGRP-CorruptedFileIT]
   > Caused by: java.lang.AssertionError: shard term already update.  op term [2], shardTerm [3]
   >    at __randomizedtesting.SeedInfo.seed([172540A0C77C8F60]:0)
   >    at org.elasticsearch.index.shard.IndexShard.lambda$acquireReplicaOperationPermit$9(IndexShard.java:2234)
   >    at org.elasticsearch.index.shard.IndexShardOperationPermits.doBlockOperations(IndexShardOperationPermits.java:173)
   >    at org.elasticsearch.index.shard.IndexShardOperationPermits.blockOperations(IndexShardOperationPermits.java:110)
   >    at org.elasticsearch.index.shard.IndexShard.acquireReplicaOperationPermit(IndexShard.java:2233)
   >    at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncReplicaAction.doRun(TransportReplicationAction.java:616)
   >    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
   >    at org.elasticsearch.action.support.replication.TransportReplicationAction$ReplicaOperationTransportHandler.messageReceived(TransportReplicationAction.java:493)
   >    at org.elasticsearch.action.support.replication.TransportReplicationAction$ReplicaOperationTransportHandler.messageReceived(TransportReplicationAction.java:479)
   >    at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:63)
   >    at org.elasticsearch.transport.TcpTransport$RequestHandler.doRun(TcpTransport.java:1679)
   >    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:723)
   >    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
   >    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
   >    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
   >    at java.lang.Thread.run(Thread.java:748)
```

**Comment by jasontedor** [comment]
Note this happened in #32118 too.

**Comment by ywelsch** [comment]
Closed by #32442. If this still occurs, please reopen.
